c++/src/kj/units.h

```diff
@@ -347,21 +374,31 @@ class Quantity {
 
   template <typename Number1, typename Number2, typename Unit2>
   friend inline constexpr auto operator*(Number1 a, Quantity<Number2, Unit2> b)
-      -> Quantity<decltype(Number1(1) * Number2(1)), Unit2>;
+      -> Quantity<decltype(Number1() * Number2()), Unit2>;
 
   template <typename T>
   friend inline constexpr T unit();
```

Comment:
To make MSVC happy, I had to change this friend declaration to match the trailing return type SFINAE style below, i.e.:

```c++
template <typename T>
friend inline constexpr auto unit() -> decltype(Unit_<T>::get());
```

Otherwise MSVC complains about an "ambiguous call to overloaded function", interpreting this friend declaration as a prototype for a function different from the trailing return type overload defined below.